cve_bin_tool/version_compare.py

```diff
@@ -0,0 +1,194 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import re
+
+"""
+A class for comparing arbitrary versions of products.
+
+Splits versions up using common whitespace delimiters and also splits out letters
+so that things like openSSL's 1.1.1y type of version will work too.
+
+This may need some additional smarts for stuff like "rc" or "beta" and potentially for
+things like distro versioning.  I don't know yet.
+"""
+
+
+class CannotParseVersionException(Exception):
+    """
+    Thrown if the version doesn't comply with our expectations
+    """
+
+
+def parse_version(version_string: str):
+    """
```

Comment:
What is packagin.version.parse missing so that you can't use that instead?

Reply:
It intentionally no longer handles anything that doesn't comply with pep440 so the errors we saw were:

- can't handle openssl or libjpeg (possibly others, although we don't have bug reports for those)
- can't handle any separator other than . (as you see, we also regularly see `_` and `-` in version numbers)
- doesn't handle cases where there are more than 3 numbers.  So our old workaround for openssl results in something like 1.1.1.5 and throwing that into packaging's version compare now gives an invalid version error.

That last one is where I gave up on trying to smoosh version numbers into pep440 compliant boxes.  There's more than a few cases where we need to compare more than 3 numbers and it seemed more risky to start multiplying them all by 1000 or something ridiculous to make it work.

Honestly, if I can get this working well enough to be usable to other people we could probably stand to throw just this file up as a version compare package on pypi for other people who need to compare the messy world of versioning.  Basically create what I was hoping cmp_version would be.